src/domains/dashboard/components/node-view/node-view.tsx

```diff
@@ -0,0 +1,307 @@
+              </div>
+            )}
+            <div className="netdata-chart-row">
+              {chartsSorted
+                .flatMap((chart) => generateHeadCharts("heads", chart, duration))
+                .map(parseChartString)
+                .map((attributes: Attributes | null) => (
+                  attributes && (
+                    <ChartWrapper
+                      attributes={attributes}
+                      key={`${attributes.id}-${attributes.dimensions}`}
+                    />
+                  )
+                ))}
+            </div>
+            {chartsSorted.map((chart) => {
+              const commonMin = chartCommonMin(chart.family, chart.context, chart.units)
+              const commonMax = chartCommonMax(chart.family, chart.context, chart.units)
+              return (
+                <div
+                  className="netdata-chartblock-container"
+                  style={{ width: `${pcentWidth}%` }}
+                  key={`${chart.id}-${chart.dimensions}`}
+                >
+                  {/* eslint-disable-next-line react/no-danger */}
+                  <span dangerouslySetInnerHTML={{
+                    __html: netdataDashboard.contextInfo(chart.context),
+                  }}
+                  />
+                  <ChartWrapper
+                    id={`chart_${name2id(chart.id)}`}
+                    attributes={{
+                      id: chart.id,
+                      chartLibrary: "dygraph",
+                      width: "100%",
+                      height: netdataDashboard.contextHeight(
+                        chart.context, options.chartsHeight,
+                      ),
+                      dygraphValueRange: netdataDashboard.contextValueRange(
+                        chart.context,
+                      ),
+                      before: 0,
+                      after: -duration,
+                      heightId: `${name2id(`${options.hostname}/${chart.id}`)}`,
+                      colors: `${netdataDashboard.anyAttribute(
+                        netdataDashboard.context, "colors", chart.context, "",
+                      )}`,
+                      decimalDigits: netdataDashboard.contextDecimalDigits(
+                        chart.context, -1,
+                      ),
+                      // add commonMin/commonMax attributes only if they are set
+                      ...(commonMin ? { commonMin } : {}),
+                      ...(commonMax ? { commonMax } : {}),
+                    }}
+                  />
+                </div>
+              )
+            })}
+          </div>
+        )
+      })}
+
+    </div>
+  )
+}
+
+
+interface Props {
+  chartsMetadata: ChartsMetadata
+}
+export const NodeView = ({
+  chartsMetadata,
+}: Props) => {
+  const [width, setWidth] = useState(0)
+  const ref = useRef<HTMLDivElement>(null)
+  useEffect(() => {
+    if (!width && ref.current) {
+      setWidth(ref.current.getBoundingClientRect().width)
```

Comment:
So, it sets correctly with single `width` as effect dependency? Not callback ref needed? I always struggle with these things.

Reply:
are you asking if it's ok to not provide `setWidth` as a effect dependency? as far as i remember the second item from `useEffect()` array is always the same (like useRef).

And i set this width only once on purpose (because that's how old Dashboard was working, we can add features later)